### PR TITLE
Documentation for App.js

### PR DESCRIPTION
**Pull request by stevenshuhyow**

### CODE REVIEW

File: client/public/js/components/App.js [hunk 1]
```diff
@@ -6,24 +6,36 @@ var React = require('react'),
     AppStore = require('../stores/AppStore');
 
 var App = React.createClass({
+  //mixins allows users to reuse code from different parts of
+  //the app even when their use cases are very different
+  //Navigation allows us to dynamically create hrefs in the render
+  //section.  Inside of mixen, we allow the entire react component
```
Comment (dmsakamoto) [inline client/public/js/components/App.js]: mixen -> mixin?

**Comment by dmsakamoto** [comment]
Found one small typo, but I am good with merging it. Just want to note it and go back later. Thanks Steven! This was helpful.

**Comment by stevenshuhyow** [comment]
np!  trying to make our lives a little easier!  I changed it!  Thanks @dmsakamoto

File: client/public/js/components/EventCreate.js [hunk 2]
```diff
@@ -33,6 +39,7 @@ var EventCreate = React.createClass({
   },
 
   // Add new event to the state and database
+  // .target redferences the dom node of e
```
Comment (ekong2) [inline client/public/js/components/EventCreate.js:42]: minor typo

File: client/public/js/components/EventCreate.js [hunk 3]
```diff
@@ -43,13 +50,17 @@ var EventCreate = React.createClass({
   },
 
   // Update the current state with the new values in the input fields
+  // every time the user tpes a letter
```
Comment (ekong2) [inline client/public/js/components/EventCreate.js:53]: types

File: client/public/js/components/EventCreate.js [hunk 4]
```diff
@@ -43,13 +50,17 @@ var EventCreate = React.createClass({
...
+  // this updates the state values that were defined in getInitialState
   _onChange: function(e) {
     var obj = {};
+    console.log(e.target, "targeted node");
     obj[e.target.name] = e.target.value;
     this.setState(obj);
   },
 
   // Event handler for 'create' events coming from the EventStore
+  // clears default calues of name and location
```
Comment (ekong2) [inline client/public/js/components/EventCreate.js:63]: values

**Comment by ekong2** [comment]
:+1:  minor typos but we can go ahead

**Pull request merged**